Assets/Scripts/UIManager.cs

```diff
@@ -0,0 +1,321 @@
+
+    [SerializeField] private Image summonImage1;
+    [SerializeField] private Image summonImage2;
+    [SerializeField] private Image summonImage3;
+    [SerializeField] private Image summonImage4;
+    [SerializeField] private Image summonImage5;
+
+    [SerializeField] private Sprite ghostImage;
+    [SerializeField] private Sprite skeletonImage;
+    [SerializeField] private Sprite zombieImage;
+
+    [SerializeField] private GameObject summonIndicator1;
+    [SerializeField] private GameObject summonIndicator2;
+    [SerializeField] private GameObject summonIndicator3;
+    [SerializeField] private GameObject summonIndicator4;
+    [SerializeField] private GameObject summonIndicator5;
+
+    private int i = 0;
+
+    public static UIManager Instance
+    {
+        get;
+        private set;
+    }
+
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+        }
+    }
+
+    public void UpdateSummon(string summon, int summonCount)
+    {
+                
+        switch (i)
+        {
+            case 0:
+                if (summonIndicator1.activeSelf == false)
+                {                    
+                    UpdateSummonImage(summon, summonCount);
+                    UpdateSummonName(summon, summonCount);
+                    summonIndicator1.SetActive(true);
+                    i++;
+                }
+                break;
+            case 1:
+                if (summonIndicator2.activeSelf == false)
+                {
+                    UpdateSummonImage(summon, summonCount);
+                    UpdateSummonName(summon, summonCount);
+                    summonIndicator2.SetActive(true);
+                    i++;
+                }
+                break;
+            case 2:
+                if (summonIndicator3.activeSelf == false)
+                {
+                    UpdateSummonImage(summon, summonCount);
+                    UpdateSummonName(summon, summonCount);
+                    summonIndicator3.SetActive(true);
+                    i++;
+                }
+                break;
+            case 3:
+                if (summonIndicator4.activeSelf == false)
+                {
+                    UpdateSummonImage(summon, summonCount);
+                    UpdateSummonName(summon, summonCount);
+                    summonIndicator4.SetActive(true);
+                    i++;
+                }
+                break;
+            case 4:
+                if (summonIndicator5.activeSelf == false)
+                {
+                    UpdateSummonImage(summon, summonCount);
+                    UpdateSummonName(summon, summonCount);
+                    summonIndicator5.SetActive(true);
+                }
+                break;
+            }
+        
+    }
+
+    public void UpdateSummonImage(string summon, int summonCount)
+    {
```

Comment:
This should be private and called on the UpdateSummon method, and pass the summon name and the instantiated prefab ui image to set the sprite using the dictionary

Reply:
Done